src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -177,16 +199,16 @@ const convertFrameworkDetail = async (framework) => {
       appLaunchedTime: new Date(framework.metadata.creationTimestamp).getTime(),
       appCompletedTime: new Date(framework.status.completionTime).getTime(),
       appExitCode: completionStatus ? completionStatus.code : null,
-      appExitSpec: {}, // TODO
-      appExitDiagnostics: completionStatus ? completionStatus.diagnostics : null,
-      appExitMessages: {
+      appExitSpec: completionStatus ? generateExitSpec(completionStatus.code) : generateExitSpec(null),
+      appExitDiagnostics: generateExitDiagnostics(diagnostics),
+      appExitMessages: diagnostics ? {
         container: null,
-        runtime: null,
-        launcher: null,
-      },
-      appExitTriggerMessage: completionStatus ? completionStatus.diagnostics : null,
-      appExitTriggerTaskRoleName: null, // TODO
-      appExitTriggerTaskIndex: null, // TODO
+        runtime: extractRuntimeOutput(diagnostics),
+        launcher: extractLauncherOutput(diagnostics),
+      } : null,
+      appExitTriggerMessage: completionStatus && completionStatus.trigger ? yaml.safeDump(completionStatus.trigger) : null,
```

Comment:
>appExitTriggerMessage [](start = 6, length = 21)

Why not just use "completionStatus.trigger.message"

Reply:
Thanks, I don't notice this field